lib/presentation/ui/home/home_page.dart

```diff
@@ -185,32 +188,55 @@ class _MyHomePageState extends BaseState<HomePage> {
   }
 }
 
-class RecentListWidget extends StatelessWidget {
+class RecentListWidget extends StatefulWidget {
   final HomeBloc bloc;
 
   RecentListWidget(this.bloc);
 
+  @override
+  _RecentListWidgetState createState() => _RecentListWidgetState();
+}
+
+class _RecentListWidgetState extends State<RecentListWidget> {
+  @override
+  void initState() {
+    super.initState();
+    widget.bloc.refreshErrorStream
+        .listen((_) => _refreshController.refreshFailed());
+  }
+
+  RefreshController _refreshController =
+      RefreshController(initialRefresh: false);
+
   @override
   Widget build(BuildContext context) {
     return StreamBuilder<List<OccurrenceModel>>(
-        stream: bloc.occurrencesStream,
+        stream: widget.bloc.occurrencesStream,
         builder: (context, snapshot) {
           if (snapshot.connectionState == ConnectionState.waiting) {
             return Center(child: Text("A carregar"));
           }
 
+          _refreshController.refreshCompleted();
           if (snapshot.data != null) {
```

Comment:
This is from a previous commit, but I'd use here `snapshot.hasData` instead to make the null check more verbose.

Reply:
yes! yoy're right, sorry it's the bad habbit